sim/src/test/scala/midasexamples/TutorialSuite.scala

```diff
@@ -377,6 +377,23 @@ class TerminationF1Test extends TutorialSuite("TerminationModule") {
   1 to 10 foreach {x => runTest(backendSimulator, args = Seq("+termination-bridge-tick-rate=10", s"+seed=${x}"), shouldPass = true)}
 }
 
+class CustomConstraintsF1Test extends TutorialSuite("CustomConstraints") {
+  def readLines(filename: String): List[String] = {
+    val file = new File(genDir, s"/${filename}")
+    Source.fromFile(file).getLines.toList
+  }
+  it should s"generate synthesis XDC file" in {
+    val xdc = readLines("FireSim-generated.synthesis.xdc")
+    xdc should contain("constrain_synth1")
+    xdc should contain("constrain_synth2 [reg firesim_top/top/sim/target/CustomConstraints_/dut/r0]")
```

Comment:
Do you have any ideas for ways we could avoid hardcoding a module path here? These are liable to change.

Reply:
Yeah the hardcoded strings also make me sad.

Theoretically you could try to throw a custom annotation into the input and then check what the target of the anno got transformed into at the end of the flow, and then check whether that also shows up here in the XDC output file. Alternatively, some form of metadata could be useful, like the ability to assign a tag to something in the IR and then at the end of the compilation asking the compiler to dig up the concrete final path that tag landed on. But that's basically what the annotations already do.

Reply:
Yeah... While not perfect, I think both of those ideas are pretty reasonable. 

In the short term you can probably just wildcard the middle component of the path? The pass-level test in GoldenGate can check for complete path correctness.

Reply:
Yeah I like that approach. We only care about the prefix and suffix of the path. Let me tweak this.

Reply:
Finally figured out how to do this. So many DSLs :see_no_evil: